README.md

```diff
@@ -51,6 +51,7 @@ Prerequisites:
 * [libevent](https://libevent.org/) (`libevent-dev` in Debian); can be avoided
   if building without tests (DISABLE_TESTS=ON)
 * OpenSSL (`libssl-dev` in Debian) if building with TLS support
+* [clang-format](https://apt.llvm.org/) for code formatting
```

Comment:
Formatting isn't a build prerequisite, is it? I think it shouldn't be needed for those who use this lib as a dependency only.

I think we should require it only when contributing code changes. I suggest we make it optional and maybe not even list it under build prerequisites. If there is a *Contributing* section in the README.md or a CONTRIBUTING.md file in the repo, we can add it there instead. (CONTRIBUTING.md, if it exists, is shown automatically when you open a PR in github.)

Reply:
Good idea, will add it to a CONTRIBUTING.md